File: easy/longest_palindrome.py
def longest_palindrome(s: str) -> int:
    freq = {}
    for c in s:
        freq[c] = freq.get(c, 0) + 1
    odds = len(list(filter(lambda x: x % 2 == 1, freq.values())))
    if odds > 1:
        return len(s) + 1 - odds
    return len(s)

File: easy/longest_palindrome.go
package easy

func LongestPalindrome(s string) int {
	freq := map[rune]int{}
	for _, char := range s {
		freq[char] += 1
	}
	odds := -1
	for _, value := range freq {
		if value%2 == 1 {
			odds++
		}
	}
	if odds > 0 {
		return len(s) - odds
	}
	return len(s)
}
